src/inverseRobot.js

```diff
@@ -8,6 +8,21 @@
 
 function inverseRobot(robot) {
   // write code here
+  const inverseArr = {};
+  const robotValues = Object.values(robot);
+  const robotKeys = Object.keys(robot);
+
+  for (let i = 0; i < robotValues.length; i++) {
+    inverseArr[robotValues[i]] = robotKeys[i];
+  }
+
+  const inverseArrKeys = Object.keys(inverseArr);
+
+  if (robotKeys.length !== inverseArrKeys.length) {
+    return null;
+  }
+
+  return inverseArr;
```

Comment:
This solution works, but let's try to do it another way. You can use for .. in loop to get keys of object, than you can access its props as object[key].

Reply:
hello i also noticed you had in checklist best example 
 "EVEN BETTER EXAMPLE if (robot.hasOwnProperty(key)) {... "
 
 but i could not figure out how to use it here, can you help me ?